Objects/stringlib/eq.h

```diff
@@ -4,14 +4,20 @@
  * unicode_eq() is called when the hash of two unicode objects is equal.
  */
 Py_LOCAL_INLINE(int)
-unicode_eq(PyObject *a, PyObject *b)
+unicode_eq(PyObject *str1, PyObject *str2)
 {
-    if (PyUnicode_GET_LENGTH(a) != PyUnicode_GET_LENGTH(b))
+    Py_ssize_t len = PyUnicode_GET_LENGTH(str1);
+    if (PyUnicode_GET_LENGTH(str2) != len) {
         return 0;
-    if (PyUnicode_GET_LENGTH(a) == 0)
-        return 1;
-    if (PyUnicode_KIND(a) != PyUnicode_KIND(b))
+    }
+
+    int kind = PyUnicode_KIND(str1);
+    if (PyUnicode_KIND(str2) != kind) {
         return 0;
-    return memcmp(PyUnicode_1BYTE_DATA(a), PyUnicode_1BYTE_DATA(b),
-                  PyUnicode_GET_LENGTH(a) * PyUnicode_KIND(a)) == 0;
+    }
+
+    const void *data1 = PyUnicode_DATA(str1);
+    const void *data2 = PyUnicode_DATA(str2);
+    int cmp = memcmp(data1, data2, len * kind);
+    return (cmp == 0);
```

Comment:
```suggestion
    return memcmp(data1, data2, len * kind) == 0;
```
and same in unicode_compare_eq()?

Reply:
Ok, I removed the *cmp* variable.